From 70d4caefed009fb698c734d745053d080c66f046 Mon Sep 17 00:00:00 2001
From: Hsieh Hung-En <hungen3108@gmail.com>
Date: Sat, 31 Jan 2026 00:00:16 +0800
Subject: [PATCH 007/385] UPSTREAM: ASoC: es8328: Use cached regmap on resume

Use the regmap stored in the driver private data when restoring the
register cache on resume, instead of looking it up from the device.

This keeps the resume path consistent with the regmap instance used by
the driver and avoids relying on a separate dev_get_regmap() lookup.

Signed-off-by: Hsieh Hung-En <hungen3108@gmail.com>
Link: https://patch.msgid.link/20260130160017.2630-5-hungen3108@gmail.com
Signed-off-by: Mark Brown <broonie@kernel.org>

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 sound/soc/codecs/es8328.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/sound/soc/codecs/es8328.c b/sound/soc/codecs/es8328.c
index a18c3fe22da5a..7afc97c62587e 100644
--- a/sound/soc/codecs/es8328.c
+++ b/sound/soc/codecs/es8328.c
@@ -771,12 +771,9 @@ static int es8328_suspend(struct snd_soc_component *component)
 
 static int es8328_resume(struct snd_soc_component *component)
 {
-	struct regmap *regmap = dev_get_regmap(component->dev, NULL);
-	struct es8328_priv *es8328;
+	struct es8328_priv *es8328 = snd_soc_component_get_drvdata(component);
 	int ret;
 
-	es8328 = snd_soc_component_get_drvdata(component);
-
 	ret = clk_prepare_enable(es8328->clk);
 	if (ret) {
 		dev_err(component->dev, "unable to enable clock\n");
@@ -790,8 +787,8 @@ static int es8328_resume(struct snd_soc_component *component)
 		return ret;
 	}
 
-	regcache_mark_dirty(regmap);
-	ret = regcache_sync(regmap);
+	regcache_mark_dirty(es8328->regmap);
+	ret = regcache_sync(es8328->regmap);
 	if (ret) {
 		dev_err(component->dev, "unable to sync regcache\n");
 		return ret;
-- 
2.52.0

